### ISSUE TITLE
Do not prefix log output

### ISSUE THREAD

**Issue by jpzwarte** [vercel/turborepo#901]
### What version of Turborepo are you using?

1.1.7-canary.5

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Mac

### Describe the Bug

<img width="805" alt="Screenshot 2022-03-18 at 15 37 50" src="https://user-images.githubusercontent.com/3968/159023437-19ff15fe-771a-491b-b147-c1814b39965c.png">

All the task log lines are prefixed with the scope, causing the logs to be hard to read.

### Expected Behavior

Please do not prefix every log line.

### To Reproduce

turbo run build --scope=foo

**Comment by gsoltis** [comment]
This might be a good candidate for another option for [log behavior](https://github.com/vercel/turborepo/blob/fc89d554987ceb7c5d20b3140e34347c5e498744/cli/internal/run/run.go#L75)

**Comment by Kingwl** [comment]
> This might be a good candidate for another option for [log behavior](https://github.com/vercel/turborepo/blob/fc89d554987ceb7c5d20b3140e34347c5e498744/cli/internal/run/run.go#L75)

What about UI or something like `cache hit` or `cache miss`?
And should we strip prefix if cache log in full mode but replay logs in content only (the new mode).

**Comment by sprohaszka** [comment]
Hi!
I am willing to help you implement this feature :)

I have some difficulties to deep dive in the code base and know where we can remove the `prefix`. I tried to rough changing this [line](https://github.com/vercel/turborepo/blob/dfc72ce349930013fd88e6d042f7b5da2781eb44/cli/internal/run/run.go#L813) without any success.

Can you help me by pointing me out where to start? :)

**Comment by rafaeltab** [comment]
I can also pick this up, or help out if necessary.

**Comment by sprohaszka** [comment]
@gsoltis I tried to do something [here](https://github.com/vercel/turborepo/pull/1349).
I may miss some modification to improve the PR ^^

**Comment by sakekasi** [comment]
Just wanted to bump this issue. I'm using a custom pino logger that outputs JSON, and my parser gets tripped up by all the log prefixes

**Comment by sprohaszka** [comment]
If it make sense, I still work on my PR to improve it.

**Comment by sakekasi** [comment]
I'd definitely appreciate that!

**Comment by sprohaszka** [comment]
> I'd definitely appreciate that!

I need to know what to improve 🙂
I did some modification last time. I don't know if it is enough 🙂

**Comment by rafaeltab** [comment]
@sprohaszka I believe, though you have changed the code, you need to resolve the requested changes from gsoltis.
Normally there is a button somewhere to do this on the requested change.

**Comment by sakekasi** [comment]
FWIW, piping the output into this sed command:

```
sed -l -E 's/^@[a-zA-Z-]+\/[a-z-]+:[a-z-]+: //'
```

did the trick for me!

**Comment by nfantone** [comment]
What about not changing the logs at all? Would that be a possibility? Just forward the original stream to `stdout`.

Right now, by using `turbo` to run commands, we lose things like colouring, formatting and hyperlinks.

<img width="627" alt="image" src="https://user-images.githubusercontent.com/476069/205642188-94ae08aa-4283-4668-8a93-477586dfe488.png">

**Comment by jrolfs** [comment]
This was one of the few (only?) things I could not find an analog for in Turborepo when migrating from Lerna (which has [`--no-prefix`](https://github.com/lerna/lerna/tree/main/commands/run#--no-prefix)). I prefer prefixed output by default for local development, but in GitHub Actions the prefix prevents annotations from working correctly.

**Comment by rafaeltab** [comment]
I see this still has not been implemented, could I pick this up?

If so are the following requirements for this correct?
- [ ] Stop adding prefixes to the stored logs.
- [ ] Add an `output-logs` flag called `no-prefix`
- [ ] Add prefix to replayed logs for `full`, `hash-only`, and `new-only` (I'll make sure to double check this list when implementing)
- [ ] Only add the prefix when `output-logs != no-prefix`
- [ ] Make sure this is also supported in the `turbo.json` file

An alternative would be to add a separate flag `--no-prefix`, this would allow the combining of settings such as `new-only` with no prefix.

**Comment by nfantone** [comment]
Can't speak for others, but at least from my perspective "removing prefixes" wouldn't cut it. I'd like `turbo` to _not modify_ output at all. If really needed (i.e.: to detail currently running process in parallel executions), it could maybe _add_ log lines of its own instead of hi-jacking others?

**Comment by rafaeltab** [comment]
I personally don't know to what extent turbo removes coloring, meta characters, etc.
I know coloring can be forced with `--color`, other than that, I'll have to find it out during implementation.

**Comment by anthonyshew** [comment]
I'm seeing interest for:
- removing log prefixes
- wanting stdout

In my estimation, making an output mode for stdout covers both concerns and with one swoop. I think that's the ticket!

@Rafaeltab, your point to being able to use this new output mode along with others is great. What do we think of the new list of options being:

- full | Displays all turbo output
- hash-only | Show only the hashes of the tasks
- new-only | Only show turbo output from cache misses
- stdout-full | Displays all stdout
- stdout-new-only | Only show stdout from cache misses
- none

**Comment by zomars** [comment]
Yes please! This is also suppressing eslint/tsc annotations when running in Github actions.

**Comment by domdomegg** [comment]
@JannesMeyer @zomars You can use [tsc-absolute](https://www.npmjs.com/package/tsc-absolute) and the suggested problem matcher setup in [its README](https://github.com/domdomegg/tsc-absolute#github-problem-matching) to get TSC annotations appearing correctly when using turborepo.

Note that because TSC does not provide absolute paths, when running in a turborepo style setup GitHub is unable to resolve the paths, so you'll need to use tsc-absolute anyways.

**Comment by marcello3d** [comment]
did this regress? the option doesn't seem to work with `turbo 2.0.10`. specifically I'm running:
```
turbo --log-prefix=none --log-order=stream run foo
```

**Comment by anthonyshew** [comment]
@marcello3d We currently have a known bug where putting flags before the command (e.g. `run`) get ignored. #8893 is meant to address this.
